File: app/src/main/java/com/cyl/musiclake/ui/music/local/contract/SongsContract.java
package com.cyl.musiclake.ui.music.local.contract;

import com.cyl.musiclake.data.db.Music;
import com.cyl.musiclake.base.BaseContract;

import java.util.List;

public interface SongsContract {

    interface View extends BaseContract.BaseView {
        void showSongs(List<Music> songList);

        void setEmptyView();
    }

    interface Presenter extends BaseContract.BasePresenter<View> {
        void loadSongs(String action);

    }
}

File: app/src/main/java/com/cyl/musiclake/ui/music/local/contract/SongsContract.kt
package com.cyl.musiclake.ui.music.local.contract

import com.cyl.musiclake.ui.base.BaseContract
import com.music.lake.musiclib.bean.BaseMusicInfo

interface SongsContract {

    interface View : BaseContract.BaseView {
        fun showSongs(songList: MutableList<BaseMusicInfo>)

        fun setEmptyView()
    }

    interface Presenter : BaseContract.BasePresenter<View> {
        fun loadSongs(isReload: Boolean)
    }
}
